### DISCUSSION TITLE
Multiple JSON inputs for opa eval command

### DISCUSSION THREAD

**Discussion by Sumanth-Tarimala** [open-policy-agent/community#406] [OPA and Rego]
The requirement is to give two JSON inputs for a rego query. Below is the command used to evaluate the policy and getting data file (input2.json) in the output. 

opa eval -d my_policy.rego -i input1.json -d input2.json data 

Please suggest is there a way to opt out a data file (input2.json) from the output. 

![image](https://user-images.githubusercontent.com/92350625/233844420-8e825785-66a2-4b32-8039-eaeeef4ac35e.png)

**Comment by anderseknert** [comment]
What do you mean by "the requirement"? Would you expect both inputs to be merged? Or two different evaluations to be performed using each of the input files in sequence?

If it's the former, you're probably best off merging the input beforehand, using `jq` or even `opa eval` with some `object.union` trickery. 

If it's the latter, [conftest](https://github.com/open-policy-agent/conftest) might be of interest.

**Reply by Sumanth-Tarimala** [reply]
I am trying to compare values from different input JSON files.  Could you please suggest a way.

**Reply by anderseknert** [reply]
Conftest should let you do that. If you want to use `opa eval`, you'd need to somehow merge the documents into one beforehenad, i.e. something like:

```json
{
  "doc1": { ... }
  "doc2": { ... }
}
```

**Reply by srenatus** [reply]
I might be missing something -- can't we also do this?
```
opa eval -d doc1:input1.json -d doc2:input2.json -d policy.rego <your query>
```

And have your policy use `data.doc1` for the first and `data.doc2` for the second input...? 🤔

**Reply by anderseknert** [reply]
Of course. But it's not using two _inputs_ specifically. The outcome would be the same though, so why not :)
